src/metrics.rs

```diff
@@ -85,64 +89,137 @@ pub fn metrics_to_string() -> std::result::Result<String, DshError> {
 /// #[tokio::main]
 /// async fn main() {
 ///    tokio::spawn(async move {
-///       start_http_server(8080).await;
+///       start_http_server(8080).await.unwrap();
 ///   });
 /// }
 /// ```
-pub async fn start_http_server(port: u16) {
-    let metrics_route = warp::path!("metrics").and_then(http_metric_response);
-    warp::serve(metrics_route).run(([0, 0, 0, 0], port)).await;
+///
+pub async fn start_http_server(port: u16) -> Result<()> {
```

Comment:
As we now already introduced Tokio dependency for spawning task for the service_fn. 

Maybe it makes sense to directly start the http server in a tokio task as well, with this the start_http_server doesn't have to be async anymore, as the whole htttpserver runs on a separate thread. With this, the user only have to call the function and not start his own tokio task, like:

``` rust
#[tokio::main]
async fn main() {
    start_http_server(8080);
}
```